xc/xenops_server_xen.ml

```diff
@@ -2082,7 +2085,11 @@ module VBD = struct
 						)
 						(fun () ->
 							Opt.iter (fun domid ->
-								Storage.dp_destroy task (Storage.id_of (string_of_int domid) vbd.Vbd.id)
+								Opt.iter
```

Comment:
I agree. Please make the change, @kc284 - and put a comment in each of the clauses to explain what's going on.